src/test/java/uk/gov/justice/digital/hmpps/vsip/services/clients/PrisonVisitsTestingHelperClient.java

```diff
@@ -199,6 +202,96 @@ public void deleteApplication(String applicationReference) {
     }
 
 
+    public String createSessionTemplate(
+            String prisonCode,
+            LocalDateTime sessionStartDateTime,
+            Integer weeklyFrequency,
+            Integer closedCapacity,
+            Integer openCapacity,
+            String locationLevels,
+            String incentive,
+            String category,
+            Boolean disableAllOtherSessionsForSlotAndPrison
+    ) {
+        final var uri = "/test/prison/"+prisonCode+"/add/template";
+        LOG.debug("Enter createSessionTemplate " + uri);
+
+        Function<ClientResponse, Mono<? extends Throwable>> statusHandler = response -> {
+            var message = "Error createSessionTemplate failed :" + uri + " http_status:" + response.statusCode();
+            LOG.error(message);
+            return Mono.error(new AssertionError(message));
+        };
+
+        var response = webClient.put()
+                .uri(builder ->
+                        getCreateSessionTemplateUri(prisonCode,
+                                sessionStartDateTime,
+                                weeklyFrequency,
+                                closedCapacity,
+                                openCapacity,
+                                locationLevels,
+                                incentive,
+                                category,
+                                disableAllOtherSessionsForSlotAndPrison,
+                                builder,
+                                uri))
+                .retrieve()
+                .onStatus(validateCreateStatusHandler, statusHandler)
+                .bodyToMono(String.class);
+
+        try {
+            String results = response.block(Duration.ofSeconds(apiTimeout));
+            LOG.debug("put  createSessionTemplate:" + results);
+            return results;
+        } catch(Exception e) {
+            LOG.error("Error put createSessionTemplate failed :" + uri, e);
+            throw e;
+        }
+    }
+
+    public void deleteSessionTemplate(String sessionTemplateReference) {
+
+        final var uri = "/test/template/" + sessionTemplateReference + "/delete";
```

Comment:
Could this be:

        final var uri = "/test/template/" + sessionTemplateReference;

and then change the webClient.put to a webClient.delete ? and have the endpoint on the helper-api be a @ DeleteMapping ?

Reply:
It's not just delete, it also updates the DB and enables and disables things.